### PR TITLE
Use ajv and use default

### PR DESCRIPTION
**Pull request by kkontosis**

### CODE REVIEW

**Comment by kkontosis** [comment]
passes library's tests

**Comment by nicolasroger17** [comment]
What does this add?

**Comment by kkontosis** [comment]
ajv (with useDefault option enabled) instead of z-schema as validator

File: lib/helpers.js [hunk 1]
```diff
@@ -27,60 +27,151 @@
...
+      lastValidator = validator;
+      lastValue = v;
+      return validator(v);
+    },
+    getLastErrors: function() {
+      const errors = lastValidator.errors;
+      if (!errors) return null;
+      _.forEach(errors, (e) => {
+        // change format of the errors to have `path` property
+        e.path = e.schemaPath.split('#/').slice(-1)[0].split('/').slice(1);
```
Comment (nicolasroger17) [inline lib/helpers.js]: I would add a before/after example.

File: lib/helpers.js [hunk 2]
```diff
@@ -27,60 +27,151 @@
...
+        if (e.keyword === e.path.slice(-1)[0]) e.path = e.path.slice(0, -1);
```
Comment (nicolasroger17) [inline lib/helpers.js]: What is this doing ?

File: lib/helpers.js (hunk 1 above)
Comment (kkontosis) [reply lib/helpers.js]: I'll add

File: lib/helpers.js (hunk 2 above)
Comment (kkontosis) [reply lib/helpers.js]: I'll add a comment.
Ajv returns this type of error path:
`{String} #/properties/fieldName/enum`

While z-schema for the same scenario returns:
`{Array} [fieldName]`

So, I'm formatting it into the z-schema format which the library expects

**Comment by nicolasroger17** [comment]
@kkontosis If you merge into develop, you will need a PR in `e-framework` to change the branch.

**Pull request merged**